### PR TITLE
Add after_session_confirm hook after session confirmation

### LINKED ISSUE #213: Add `after_session_confirm` hook after session confirmation
Just a suggestion to add a `after_session_confirm` similar to `after_session_save`. This would let folks mark accounts as "verified" or "has_verified_email" after session confirmation.

### PR DESCRIPTION
**Pull request by wilburhimself**
fixes #213 

This PR introduces a new `after_session_confirm` hook to the Passwordless gem. This feature allows users to perform custom actions after a session has been successfully confirmed, such as marking an email as verified or updating user attributes after their first successful login.

## Changes
- Added `after_session_confirm` configuration option
- Updated `SessionsController` to call the hook after successful authentication
- Added new tests in `SessionsControllerTest` to cover the new functionality
- Updated README with documentation for the new feature

## Motivation
Users of the gem often need to perform specific actions after a successful login, such as updating user attributes or triggering other processes. The `after_session_confirm` hook provides a clean, configurable way to achieve this without modifying the gem's core code.

## Implementation Details
- The hook is called in the `authenticate_and_sign_in` method of `SessionsController`, right after successful authentication but before redirecting the user.
- The hook can be configured to accept either one argument (the session) or two arguments (the session and the request), providing flexibility for different use cases.
- If the hook is not defined or doesn't respond to `call`, it is silently ignored to maintain backwards compatibility.

## How to Use
Users can configure the hook in their Passwordless configuration:

```ruby
Passwordless.configure do |config|
  config.after_session_confirm = ->(session) {
    user = session.authenticatable
    user.update(email_verified: true)
  }
end
```

## Testing
New tests have been added to `SessionsControllerTest` to ensure the hook is called correctly in various scenarios:
- Successful authentication
- Hook receiving both session and request objects
- Hook not being called on failed authentication

All existing tests pass with these new changes.

## Reviewer Notes
- Please pay special attention to the placement of the hook call in `SessionsController` to ensure it's in the most appropriate location.
- Consider if any additional error handling is needed around the hook execution.
- Let me know if you think any additional tests or documentation would be beneficial.

## Checklist
- [x] Code follows the style guidelines of the project
- [x] New tests added for the feature
- [x] All tests pass
- [x] Documentation (README) has been updated

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -243,6 +243,51 @@ Passwordless.configure do |config|
   config.sign_out_redirect_path = '/' # After a user signs out
 
   config.paranoid = false # Display email sent notice even when the resource is not found.
+
+  config.after_session_confirm = ->(session) {} # Called after a session is confirmed.
```
Comment (mikker) [inline README.md:247]:
```suggestion
  config.after_session_confirm = ->(request, session) {} # Called after a session is confirmed.
```

Let's just always send req + session in this. The other one is mostly there to keep the api backwards compatible.

File: README.md [hunk 2]
```diff
@@ -264,6 +266,24 @@ Passwordless.configure do |config|
...
+
+  config.after_session_confirm = ->(session, request) {
+    # Your custom logic here
+    user = session.authenticatable
+    user.update(email_verified: true)
+    user.update(last_login_ip: request.remote_ip)
+  }
+end
+```
+
```
Comment (mikker) [inline README.md:286]:
```suggestion
## After Session Confirm Hook

An `after_session_confirm` hook is called after a successful session confirmation – in other words: after a user signs in successfully.

```ruby
Passwordless.configure do |config|
  config.after_session_confirm = ->(session, request) {
    user = session.authenticatable
    user.update!(
      email_verified: true.
      last_login_ip: request.remote_ip
    )
  }
end
```

```

File: lib/passwordless/config.rb [hunk 3]
```diff
@@ -51,6 +51,12 @@ class Configuration
 
     option :paranoid, default: false
 
+    option(
+      :after_session_confirm,
+      default: lambda do |session, _request|
```
Comment (mikker) [inline lib/passwordless/config.rb:56]:
```suggestion
      default: lambda do |_session, _request|
```

**Comment by mikker** [comment]
Failing test is just `rails#main`

**Comment by mikker** [comment]
Great work! ❤️🧡💛💚💙💜

**Comment by wkirby** [comment]
@mikker when can we expect to see this released?

**Comment by mikker** [comment]
Just pushed 1.8.0 with this change 😊

**Comment by henrikbjorn** [comment]
Am I correct in assuming the controller context is lot in the hook ?

A use case would be for Ahoy to call `ahoy.authenticate` for analytics. But it seems that can't be done since the hook is "just" a function.

Would it make sense to instead define the hook methods on the controller (kind of like Devise)? Then the Controller context is kept, and it is also easy to set a custom controller to use.

**Comment by henrikbjorn** [comment]
Another note. It is not possible to have two different hooks. E.g one for AdminUsers and one for normal Users. Which having it as a controller method would handle.

**Comment by mikker** [comment]
Good points Henrik!

It could be useful to have the controller context at hand.

Maybe something like if setting the hook setting to a `:symbol`, a method with that name will be called instead?

Like

```ruby
class ApplicationController < ...
  private

  def pwless_after_session_confirm(session) # no need to pass request either
    # ...
  end
end

Passwordless.configure do |config|
  config.after_session_confirm = :pwless_after_session_confirm
end
```

That would make it a non-breaking change.

**Pull request merged**